### ISSUE TITLE
Page list: Pages without a title has no link text

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#54959]
### Description

In the navigation block, posts without titles have the link text `(no title)`.
But in the page list, pages without titles have no default or fallback link text. This happens for page lists inside and outside navigation blocks.

Page lists inside navigation blocks can be "detached", and by selecting the page list item, the user can add a custom "label":
<img width="276" alt="The label input field in the page list block settings sidebar" src="https://github.com/WordPress/gutenberg/assets/7422055/729c0dca-8029-443b-815e-53360218c762">
But this is is not possible for the page list when it is not inside the navigation block.

This is a problem for users because without a link text, the link is not discoverable without using AT like a screen reader.
On the front of the website, neither mouse or keyboard users would be able to identify or access the link.

The empty link would only show as a mysterious blank space, depending on the styling of the parent block.
Navigation:
<img width="635" alt="Navigation block with space instead of a visible link " src="https://github.com/WordPress/gutenberg/assets/7422055/5781b29c-2312-4f30-ad8c-f22f8eafac5b">
Page list outside the navigation block is showing an empty list item:
<img width="191" alt="Page list with empty list item" src="https://github.com/WordPress/gutenberg/assets/7422055/33064f95-6795-4787-a8d1-a68096469e6c">

### Step-by-step reproduction instructions

1. Create a page and publish it without adding a title.
2. Place a page list block
3. Place a navigation block with a page list block inside.

### Environment info

Gutenberg current trunk.
WordPress Beta tester plugin Version 3.5.4.
Only tested on 6.4 beta 1 current nightly: I did not test on older versions of WordPress.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

**Comment by carolinan** [comment]
This was discovered while doing an accessibility-ready review of Twenty Twenty-Four, which includes a page template called "page no title".
While testing this template, I did not add a title to the page because I thought it would not be needed. As a user I did not consider that the title may be needed elsewhere on the site.

**Comment by carolinan** [comment]
I had a very quick look at WordPress core and I found that the nav menu functions adds the fall back title for posts,
and the page related walkers adds the fall back title for pages, so I believe this is a problem with a missing feature in the block, not core.

https://developer.wordpress.org/reference/classes/walker_page/
https://developer.wordpress.org/reference/classes/walker_pagedropdown/
